oodeel/methods/dknn.py

```diff
@@ -78,15 +78,15 @@ def _score_tensor(self, inputs: TensorType) -> np.ndarray:
             inputs: input samples to score
 
         Returns:
-            scores
+            tuple: scores and dictionary containing logits and labels.
         """
 
-        input_projected = self.feature_extractor(inputs)
+        input_projected, info = self.feature_extractor.predict(inputs)
         input_projected = self.op.convert_to_numpy(input_projected)
         input_projected = input_projected.reshape(input_projected.shape[0], -1)
         norm_input_projected = self._l2_normalization(input_projected)
         scores, _ = self.index.search(norm_input_projected, self.nearest)
-        return scores[:, -1]
+        return scores[:, -1], info
```

Comment:
I think this function should return `scores, logits` since it has no access to the labels, and then info dict would be built in `score` function of the parent class. (same remark for all detectors)